core/src/com/testing/screens/MainGameScreen.java

```diff
@@ -5,25 +5,37 @@
 import com.testing.BaseScreen;
 import com.testing.TestingApp;
 import com.testing.objects.TextureObject;
+import com.testing.objects.logic.GameLogic;
 
 public class MainGameScreen extends BaseScreen {
 
-    public MainGameScreen(TestingApp game) {
+
+    public MainGameScreen(TestingApp game, int level) {
         super(game);
 
-        background = new TextureObject("screen_bg/main_game_screen_bg");
+        // TODO: this is not balanced at all and is only used to check if level selection affects the levels (it does)
+        GameLogic gameLogic = new GameLogic((level / 2) + 3, (level / 4) + 2);
+        objects.add(gameLogic);
+        objects.addAll(gameLogic.getRects()); // this was not intentional
+        objects.add(gameLogic.getSequence());
+

```

Comment:
rects and sequence should be handled by gameLogic, not added to the screen object list

Reply:
`RectangleButton` uses `ObjectUpdate()` which is protected and cannot be called from `GameLogic`
 `Sequence` is no longer a `BaseObject` as requested [here](https://github.com/tacticalhippopotamus/testing-app/pull/31#discussion_r593394144)

Reply:
`ObjectUpdate()` is protected as it is handled by the generic `Object.update()` method. 
In the `GameLogic.updateDrawable()` method, `RectangleButton.update()` should be called so the buttons update when the gamelogic does.

Reply:
Sure, but that would mean that `GameLogic` has to have access to the batch. 
I think this is fine for now, it should get resolved when we do the refactor. see [this](https://github.com/tacticalhippopotamus/testing-app/pull/31/commits/3cf6687da0acb13a28f709e123f523d9de6dff6f) commit

Reply:
1. I agree that a global spritebatch is a good idea and should be done in the refactor
2. GameLogic should implement OTIDrawable because it has drawn attributes (the buttons)
3. Because GameLogic would then implement updateDrawable(), it would have access to the spritebatch

Reply:
Done, also yes, probably should have a refactor before the next build